src/Webservices/Profile.php

```diff
@@ -0,0 +1,90 @@
+<?php
+/**
+ * Profile class.
+ *
+ * Profiles are like schemas and define a kind of template which is used to construct a resource
+ * object in memory.  Representations can then be derived from the resource object.  A profile
+ * defines the name and type of each property and could be extended with other information.
+ * An automatic process could be written to generate machine-readable profile documents, such as
+ * ALPS or JSON-LD contexts from the profile.
+ *
+ * New data types can be added by adding a class file in the Type directory.  The new type can
+ * then be referenced in the profile.
+ */
+
+namespace Joomla\Webservices\Webservices;
+
+class Profile
```

Comment:
This class has been somewhat neglected in the refactoring so far, but there are quite a few methods in the main webservices class that belong here.  I just haven't got around to moving them yet.  The comment in the final method is really about splitting the XML files into smaller chunks.  For example, there is no point loading and parsing a huge XML file containing information on how to handle at least 4 different operations when a single request can only ever map to a single operation.

The current codebase tends to deal directly with XML far too much and the main configuration XML files actually perform a number of quite different functions which just adds to the confusion.  The intention here is to abstract the functions that deal with "profiles" from the other stuff.  The main codebase shouldn't be concerned with how and where the profile data is stored; it should just call methods in this class to get the information needed.